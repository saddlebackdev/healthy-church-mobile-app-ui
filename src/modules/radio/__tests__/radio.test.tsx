// npx jest ./src/modules/radio/__tests__/radio.test.ts

import * as React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import Radio from '../radio';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

describe('Radio', () => {
  let onChangeMock: any, options: any;

  beforeEach(() => {
    onChangeMock = jest.fn();

    options = [
      {
        label: 'Option 1',
        value: 1,
        testID: 'radio-option-1',
        accessibilityLabel: 'radio-option-1',
      },
      {
        label: 'Option 2',
        value: 2,
        testID: 'radio-option-2',
        accessibilityLabel: 'radio-option-2',
        disabled: true,
      },
      {
        label: 'Option 3',
        value: 3,
        testID: 'radio-option-3',
        accessibilityLabel: 'radio-option-3',
      },
    ];
  });

  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <Radio selected={1} options={options} onChange={onChangeMock} />
        </ThemeProvider>,
      );

      const radio = wrapper.getByTestId('radio');

      expect(radio.children).toHaveLength(options.length);

      expect(radio).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('renders the children correctly', () => {
      const {getByText} = render(
        <ThemeProvider theme={defaultTheme}>
          <Radio selected={1} options={options} onChange={onChangeMock} />
        </ThemeProvider>,
      );

      const option1 = getByText(options[0].label);
      const option2 = getByText(options[1].label);
      const option3 = getByText(options[2].label);

      expect(option1).toBeDefined();
      expect(option2).toBeDefined();
      expect(option3).toBeDefined();
    });

    describe('Variants', () => {
      it('vertical: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Radio
              selected={1}
              options={options}
              onChange={onChangeMock}
              direction="vertical"
            />
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });

      it('horizontal: should match snapshot', () => {
        const wrapper = render(
          <ThemeProvider theme={defaultTheme}>
            <Radio
              selected={1}
              options={options}
              onChange={onChangeMock}
              direction="horizontal"
            />
          </ThemeProvider>,
        );

        expect(wrapper).toMatchSnapshot();
      });
    });
  });

  describe('Interactions', () => {
    it('should call onChange function on click of radio button', () => {
      const {getByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <Radio selected={1} options={options} onChange={onChangeMock} />
        </ThemeProvider>,
      );

      const button = getByTestId('radio-option-1');

      fireEvent.press(button);

      expect(onChangeMock).toHaveBeenCalledTimes(1);
    });

    it('should not call onChange function on click of radio button when option is disabled', () => {
      const {getByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <Radio selected={1} options={options} onChange={onChangeMock} />
        </ThemeProvider>,
      );

      const button = getByTestId('radio-option-2');

      fireEvent.press(button);

      expect(onChangeMock).not.toHaveBeenCalled();
    });
  });
});
