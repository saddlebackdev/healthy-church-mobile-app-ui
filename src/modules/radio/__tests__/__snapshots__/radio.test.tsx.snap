// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio Rendering Variants horizontal: should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
  testID="radio"
>
  <View
    accessibilityLabel="radio-option-1"
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 0,
        "marginRight": 22,
        "opacity": 1,
      }
    }
    testID="radio-option-1"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#DBE0E3",
            "borderRadius": 22,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 22,
            "justifyContent": "center",
            "width": 22,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#00AEFF",
              "borderRadius": 22,
              "height": 14,
              "justifyContent": "center",
              "width": 14,
            },
          ]
        }
      />
    </View>
    <Text
      accessibilityLabel="radio-option-1-label"
      style={
        Array [
          Object {
            "color": "#1C2530",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingLeft": 11,
            "paddingTop": 2,
            "textAlign": "left",
          },
        ]
      }
      testID="radio-option-1-label"
    >
      Option 1
    </Text>
  </View>
  <View
    accessibilityLabel="radio-option-2"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 0,
        "marginRight": 22,
        "opacity": 1,
      }
    }
    testID="radio-option-2"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#DBE0E3",
            "borderRadius": 22,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 22,
            "justifyContent": "center",
            "width": 22,
          },
        ]
      }
    />
    <Text
      accessibilityLabel="radio-option-2-label"
      muted={true}
      style={
        Array [
          Object {
            "color": "#97A4AB",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingLeft": 11,
            "paddingTop": 2,
            "textAlign": "left",
          },
        ]
      }
      testID="radio-option-2-label"
    >
      Option 2
    </Text>
  </View>
  <View
    accessibilityLabel="radio-option-3"
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 0,
        "marginRight": 0,
        "opacity": 1,
      }
    }
    testID="radio-option-3"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#DBE0E3",
            "borderRadius": 22,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 22,
            "justifyContent": "center",
            "width": 22,
          },
        ]
      }
    />
    <Text
      accessibilityLabel="radio-option-3-label"
      style={
        Array [
          Object {
            "color": "#1C2530",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingLeft": 11,
            "paddingTop": 2,
            "textAlign": "left",
          },
        ]
      }
      testID="radio-option-3-label"
    >
      Option 3
    </Text>
  </View>
</View>
`;

exports[`Radio Rendering Variants vertical: should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
      },
    ]
  }
  testID="radio"
>
  <View
    accessibilityLabel="radio-option-1"
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 22,
        "marginRight": 0,
        "opacity": 1,
      }
    }
    testID="radio-option-1"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#DBE0E3",
            "borderRadius": 22,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 22,
            "justifyContent": "center",
            "width": 22,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#00AEFF",
              "borderRadius": 22,
              "height": 14,
              "justifyContent": "center",
              "width": 14,
            },
          ]
        }
      />
    </View>
    <Text
      accessibilityLabel="radio-option-1-label"
      style={
        Array [
          Object {
            "color": "#1C2530",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingLeft": 11,
            "paddingTop": 2,
            "textAlign": "left",
          },
        ]
      }
      testID="radio-option-1-label"
    >
      Option 1
    </Text>
  </View>
  <View
    accessibilityLabel="radio-option-2"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 22,
        "marginRight": 0,
        "opacity": 1,
      }
    }
    testID="radio-option-2"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#DBE0E3",
            "borderRadius": 22,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 22,
            "justifyContent": "center",
            "width": 22,
          },
        ]
      }
    />
    <Text
      accessibilityLabel="radio-option-2-label"
      muted={true}
      style={
        Array [
          Object {
            "color": "#97A4AB",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingLeft": 11,
            "paddingTop": 2,
            "textAlign": "left",
          },
        ]
      }
      testID="radio-option-2-label"
    >
      Option 2
    </Text>
  </View>
  <View
    accessibilityLabel="radio-option-3"
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 0,
        "marginRight": 0,
        "opacity": 1,
      }
    }
    testID="radio-option-3"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#DBE0E3",
            "borderRadius": 22,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 22,
            "justifyContent": "center",
            "width": 22,
          },
        ]
      }
    />
    <Text
      accessibilityLabel="radio-option-3-label"
      style={
        Array [
          Object {
            "color": "#1C2530",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingLeft": 11,
            "paddingTop": 2,
            "textAlign": "left",
          },
        ]
      }
      testID="radio-option-3-label"
    >
      Option 3
    </Text>
  </View>
</View>
`;

exports[`Radio Rendering renders itself correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
    ]
  }
  testID="radio"
>
  <View
    accessibilityLabel="radio-option-1"
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 22,
        "marginRight": 22,
        "opacity": 1,
      }
    }
    testID="radio-option-1"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#DBE0E3",
            "borderRadius": 22,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 22,
            "justifyContent": "center",
            "width": 22,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#00AEFF",
              "borderRadius": 22,
              "height": 14,
              "justifyContent": "center",
              "width": 14,
            },
          ]
        }
      />
    </View>
    <Text
      accessibilityLabel="radio-option-1-label"
      style={
        Array [
          Object {
            "color": "#1C2530",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingLeft": 11,
            "paddingTop": 2,
            "textAlign": "left",
          },
        ]
      }
      testID="radio-option-1-label"
    >
      Option 1
    </Text>
  </View>
  <View
    accessibilityLabel="radio-option-2"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 22,
        "marginRight": 22,
        "opacity": 1,
      }
    }
    testID="radio-option-2"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#DBE0E3",
            "borderRadius": 22,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 22,
            "justifyContent": "center",
            "width": 22,
          },
        ]
      }
    />
    <Text
      accessibilityLabel="radio-option-2-label"
      muted={true}
      style={
        Array [
          Object {
            "color": "#97A4AB",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingLeft": 11,
            "paddingTop": 2,
            "textAlign": "left",
          },
        ]
      }
      testID="radio-option-2-label"
    >
      Option 2
    </Text>
  </View>
  <View
    accessibilityLabel="radio-option-3"
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 0,
        "marginRight": 0,
        "opacity": 1,
      }
    }
    testID="radio-option-3"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#DBE0E3",
            "borderRadius": 22,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 22,
            "justifyContent": "center",
            "width": 22,
          },
        ]
      }
    />
    <Text
      accessibilityLabel="radio-option-3-label"
      style={
        Array [
          Object {
            "color": "#1C2530",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "paddingLeft": 11,
            "paddingTop": 2,
            "textAlign": "left",
          },
        ]
      }
      testID="radio-option-3-label"
    >
      Option 3
    </Text>
  </View>
</View>
`;
